SerialPrograms/Source/PokemonHome/Programs/PokemonHome_BoxSorting.cpp

```diff
@@ -117,13 +128,34 @@ struct Cursor{
   uint16_t line;
 };
 
+struct Pokemon{
+    // When adding any new member here, do not forget to modify the three operators below
+    uint16_t dex_number;
+    std::string ball_slug;
+};
 
+bool operator==(const Pokemon& lhs, const Pokemon& rhs){
+    return lhs.dex_number == rhs.dex_number &&
+        lhs.ball_slug == rhs.ball_slug;
+}
 
-void swap(std::vector<uint16_t>* arr, int pos1, int pos2){
-    int temp;
-    temp = (*arr)[pos1];
-    (*arr)[pos1] = (*arr)[pos2];
-    (*arr)[pos2] = temp;
+bool operator<(const Pokemon& lhs, const Pokemon& rhs) {
+    // For example, we could image that people want to sort first by ball type, then by dex number
+    if (lhs.ball_slug < rhs.ball_slug){
+        return true;
+    }
+    if (lhs.ball_slug > rhs.ball_slug){
+        return false;
+    }
```

Comment:
I won't commit this when merging. that way, even if it read balls, it won't change the default behaviour